blog/2022-05-04-announcing-stryker-6-effortless-sauce-mutation.md

```diff
@@ -0,0 +1,257 @@
+---
+slug: stryker-js-v6.0-effortless-sauce-mutation
+title: 'StrykerJS v6.0: Effortless Sauce Mutation'
+author: Nico Jansen
+author_title: Stryker Team
+author_url: https://github.com/nicojs
+author_image_url: https://avatars3.githubusercontent.com/u/1828233?s=400&u=fec18ad3776aaafec54c49bbd7173a841ae7ea59&v=4
+tags: [stryker]
+---
+
+We're excited to announce the next major release of the StrykerJS: StrykerJS 6.0.
+
+StrykerJS now supports _native ECMAScript modules_, or ESM for short. To make this work, we changed some parts of StrykerJS, which includes small (yet technically breaking) changes. As a bonus, we've improved performance considerably in some use cases! We're curious to know what you think. Why not try it today?
+
+<!-- truncate -->
+
+Update your Stryker dependencies to get started using the beta.
+
+```shell
+npm install --save-dev @stryker-mutator/core@latest
+# OR
+yarn add --dev @stryker-mutator/core@latest
+```
+
+Don't forget to update Stryker plugins. For example, when using the mocha runner:
+
+```shell
+npm install --save-dev @stryker-mutator/mocha-runner@latest
+# OR
+yarn add --dev @stryker-mutator/mocha-runner@latest
+```
+
+If you're new to mutation testing, it's a way to measure your tests' effectiveness. Please take a look at our RoboBar 🤖🍷 example and see how even 100% code coverage doesn't tell the whole story. Mutation testing is the only way to know if your tests are any good.
+
+Since the 6.0.0 beta release, we've implemented a new warning message to inform users about `--ignoreStatic` and improved the performance and progress reporting; you can read about that in the [🗿 Static mutant improvements](#-static-mutant-improvements) paragraph.
+
+Now, let's dive into the changes! 🏊‍♂️
+
+## 📦 ESM support
+
+As stated in the introduction, StrykerJS v6 brings official support for ESM projects.
+
+With ESM, you can use `import` and `export` statements. NodeJS has supported ESM since version 12. Browsers have gradually been adding support since 2017, and all major browsers support it by now. I recommend watching [Gil Tayar's introduction to ESM](https://www.youtube.com/watch?v=Zyoztl_7l7g) when you want to brush up on the ESM basics.
+
+The `@stryker-mutator/karma-runner` and `@stryker-mutator/jest-runner` already worked with ESM (for Jest, you will need to configure the [`--experimental-vm-modules`](../docs/stryker-js/jest-runner#ecmascript-modules) flag), but our node-based runners did not. This changes with version 6.
+
+You configure it like this:
+
+```json
```

Comment:
empty code block

Reply:
As you can see by my comment, I actually thought something was missing ;)

Reply:
Do you want me to remove the empty code block? I like it as a joke, but I hope it is clear 🤡